test/unit/providers/portion/PortionProvider.test.ts

```diff
@@ -0,0 +1,253 @@
+        });
+
+        it(`token symbol ${tokenSymbol1} to token symbol ${tokenSymbol2} within the list, should have portion`, async () => {
+          const requestedAmount = BigNumber.from(parseAmount('1.01', token1).quotient.toString())
+          const sharedInfo: QuoteRequestInfo = {
+            requestId: uuid(),
+            tokenInChainId: ChainId.MAINNET,
+            tokenOutChainId: ChainId.MAINNET,
+            tokenIn: tokenSymbol1,
+            tokenOut: tokenSymbol2,
+            amount: requestedAmount,
+            type: TradeType.EXACT_OUTPUT,
+            slippageTolerance: "0.5",
+            swapper: uuid(),
+            useUniswapX: false,
+          }
+
+          await exactOutGetPortionAndAssert(sharedInfo, token2)
+        });
+      })
+
+      async function exactOutGetPortionAndAssert(sharedInfo: QuoteRequestInfo, token2: Currency | Token) {
+        const userTypedTokenOutAmount = parseAmount('1605.56', token2);
+        const fullPortionPayload = await portionProvider.getPortionForTokenOut(sharedInfo, userTypedTokenOutAmount.quotient.toString());
+        expect(fullPortionPayload.portionResponse?.hasPortion).toBe(portionResponse.hasPortion);
+        expect(fullPortionPayload.portionResponse?.portion).toBeDefined;
+
+        if (fullPortionPayload.portionResponse?.hasPortion && fullPortionPayload.portionResponse?.portion) {
+          expect(fullPortionPayload.portionResponse?.portion.bips).toBe(portionResponse.portion.bips);
+          expect(fullPortionPayload.portionResponse?.portion.receiver).toBe(portionResponse.portion.receiver);
+          expect(fullPortionPayload.portionResponse?.portion.type).toBe(portionResponse.portion.type);
+        }
+
+        const portionAdjustedQuote = await portionProvider.getPortionAdjustedQuote(sharedInfo, userTypedTokenOutAmount.quotient.toString(), fullPortionPayload.portionAmount?.add(userTypedTokenOutAmount).quotient.toString() ?? userTypedTokenOutAmount.quotient.toString(), userTypedTokenOutAmount.quotient.toString());
+
+        // 1605.56 * 10^8 * 5 / 10000 = 80278000
+        const expectedPortionAmount = userTypedTokenOutAmount.multiply(new Fraction(portionResponse.portion.bips, 10000))
+
+        // important assertions
+        expect(portionAdjustedQuote.portionAmount?.quotient.toString()).toBe(expectedPortionAmount.quotient.toString());
+        expect(portionAdjustedQuote.portionResponse?.hasPortion).toBe(portionResponse.hasPortion);
+        expect(portionAdjustedQuote.portionResponse?.portion).toBeDefined;
+
+        if (portionAdjustedQuote.portionResponse?.hasPortion && portionAdjustedQuote.portionResponse?.portion) {
+          expect(portionAdjustedQuote.portionResponse?.portion.bips).toBe(portionResponse.portion.bips);
+          expect(portionAdjustedQuote.portionResponse?.portion.receiver).toBe(portionResponse.portion.receiver);
+          expect(portionAdjustedQuote.portionResponse?.portion.type).toBe(portionResponse.portion.type);
+        }
```

Comment:
You can just compare the portion objects of `portionAdjustedQuote.portionResponse` and  `portionResponse`